CONTRIBUTING.md

```diff
@@ -0,0 +1,66 @@
+
+### Installation
+```
+$ git clone https://github.com/wikimedia/MediaWiki-Action-API-Code-Samples.git
+$ cd MediaWiki-Action-API-Code-Samples
+Install the necessary python modules with pip
+$ python3 name_of_the_file.py #Enter any credentials if required in the file
+```
+
+### Contributing code samples
+First, propose an idea for a code sample, demo app, etc. by creating an issue around it in the repository. After discussing your idea with the repo contributors, start working, and then send a pull request, when you've your changes ready to be accepted/ merged! You can autogenerate python files for GET Requests demos where feasible by following the instructions below:
+```
+$ cd MediaWiki-Action-API-Code-Samples
+$ Add module information to `modules.json`
+$ cd python
+$ python autogenerator.py
+$ Make desired changes to the newly generated file(s)
+```
+
+## Issues
+
+If you are having difficulty after looking over your configuration carefully, please post
+a question to [StackOverflow with the mediawiki tag](http://stackoverflow.com/tags/mediawiki).
+
+**If you have discovered a bug or have a feature suggestion, please [create an issue on GitHub](https://github.com/wikimedia/MediaWiki-Action-API-Code-Samples/issues/new).**
+
+Do you want to fix an issue?  Look at the [issues](https://github.com/wikimedia/MediaWiki-Action-API-Code-Samples/issues).
+
+## Working on an Action Page
+
+* Read the project description thoroughly and the [MediaWiki Action API](https://www.mediawiki.org/wiki/API:Main_page) too.
+* Pick an action page from the [list of the action APIs](https://www.mediawiki.org/wiki/Category:MediaWiki_action_API) that you would like to review and improve.
+* Create a page in [your Sandbox](https://www.mediawiki.org/wiki/User:your_username/Sandbox/API:pagename) and do not make edits directly to the API pages.
+* Fill it with the template content which is in [here](https://www.mediawiki.org/wiki/API:Documentation_template) (copy and paste). We encourage you to stick to the template if for some reason you feel the urge to add or remove any section, discuss with your mentors.
+* For your reference, read a couple recent API pages that we modified using the new template: [API:Geosearch](https://www.mediawiki.org/wiki/API:Geosearch), [API:Parsing_wikitext](https://www.mediawiki.org/wiki/API:Parsing_wikitext)
+* Contribute sample code for the API page to [this repository](https://github.com/wikimedia/MediaWiki-Action-API-Code-Samples).
+* When you are done, get some feedback on your work on [Zulip](https://wikimedia.zulipchat.com/#narrow/stream/180873-gsoc19-outreachy18).
+
+## Submitting Changes
+
+After getting some feedbacks, push to your fork and submit a pull request. We
+may suggest some changes or improvements or alternatives, but for small changes
+your pull request should be accepted quickly.
```

Comment:
You may also add the following as we plan to avoid too many merges that we see in the git history:

_To keep the `git` history clean we are now following the rebase model. Which means, keep your branch updated against the upstream `master` branch by rebasing against it as you work on your own branch. Your commits may be squashed and merged if needed as well._